src/crimson/net/SocketMessenger.cc

```diff
@@ -161,10 +172,15 @@ seastar::future<> SocketMessenger::do_start(Dispatcher *disp)
               });
           });
       }).handle_exception_type([this] (const std::system_error& e) {
-        // stop gracefully on connection_aborted
-        if (e.code() != error::connection_aborted) {
+        // stop gracefully on connection_aborted and invalid_argument
+        if (e.code() != error::connection_aborted
+         && e.code() != error::invalid_argument) {
```

Comment:
nit, i'd suggest move `&&` to the end of previous line.

Reply:
done